src/net/discovery/mod.rs

```diff
@@ -0,0 +1,25 @@
+pub mod config;
+
+use crate::helpers::Identity;
+use crate::net::MpcHelperClient;
+use crate::protocol::prss;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error(transparent)]
+    ParseError(#[from] serde_json::Error),
+    #[error(transparent)]
+    IOError(#[from] std::io::Error),
+}
+
+/// Provides a set of peer helpers for an MPC computation. Also includes the client pointing to the
+/// running server. Since the running server is aware of which [`Identity`] it is (`H1`, `H2`, or
+/// `H3`), it should be able to use only the references to other servers. However, it's possible for
+/// a server to send data to itself.
+///
+/// Any potential failures should be captured in the initialization of the implementer.
+#[allow(clippy::module_name_repetitions)] // following standard naming convention
+pub trait PeerDiscovery {
+    fn peers(&self) -> [MpcHelperClient; 3];
+    fn prss(&self, role: Identity) -> prss::Endpoint;
```

Comment:
I think it couples too much with the implementation of PRSS and HTTP. It is no longer solves one problem of discovering peers, it functions as helper configuration that is capable of constructing things that are unrelated (http client and prss). I worry that this may lead to "god object".

I would  suggest to make this a peer discovery trait only 

```rust
   // or peer(&self, role: Identity) 
   fn peers(&self) -> [<Information that is necessary to construct http client and prss. PRSS public key, http endpoint, etc>; 3]
```

Reply:
gotcha. so return the same information, but let something else (some HTTP entity/PRSS entity) use that information to actually construct things. i think that makes sense to me